src/index.jsx

```diff
@@ -8,36 +8,43 @@ import App from './components/app';
 
 import './index.pcss';
 
-if (process.env.GA) {
-  ReactGA.initialize(process.env.GA);
-  ReactGA.pageview(window.location.pathname + window.location.search);
-
-  window.addEventListener('error', (event) => {
-    ReactGA.exception({
-      description: `ERROR: ${JSON.stringify(event, Object.getOwnPropertyNames(event))}`,
+const errorReporting = () => {
+  if (process.env.GA) {
```

Comment:
It hit me that we probably should use React hooks here so we unbind the event listeners in developer environments when the module gets replaced/reloaded.

Reply:
Guessing the listeners will only be added if GA is set, but added useEffect(...)
Would be grateful for your review again 🙏

Reply:
Close! Hooks have one caveat: they *always* has to be executed in the same order during a render, so you cannot have them executed under if-clauses and the like, instead, do the if-clause inside it, or tell React to only execute it whenever specific variables change:

```js
useEffect(() => {
  // similar to "componentDidMount" or "componentDidUpdate"
  // will execute every time `process.env.GA` changes
  return () => { /* will be executed when the component gets dismounted or the condition changed for the effect (similar to "componentDidUnmount") */ };
}, [process.env.GA]);
```